packages/svelte/src/html-tree-validation.js

```diff
@@ -137,12 +138,24 @@ const disallowed_children = {
  * Returns false if the tag is not allowed inside the ancestor tag (which is grandparent and above) such that it will result
  * in the browser repairing the HTML, which will likely result in an error during hydration.
  * @param {string} tag
- * @param {string} ancestor Must not be the parent, but higher up the tree
+ * @param {string[]} ancestors All nodes starting with the parent, up until the ancestor, which means two entries minimum
  * @returns {boolean}
  */
-export function is_tag_valid_with_ancestor(tag, ancestor) {
-	const disallowed = disallowed_children[ancestor];
-	return !disallowed || ('descendant' in disallowed ? !disallowed.descendant.includes(tag) : true);
+export function is_tag_valid_with_ancestor(tag, ancestors) {
+	const target = ancestors[ancestors.length - 1];
+	const disallowed = disallowed_children[target];
+	if (!disallowed) return true;
+
+	if ('reset_by' in disallowed && disallowed.reset_by) {
```

Comment:
We should just be able to check `disallowed.reset_by` without checking `'reset_by' in disallowed`. Is this to keep TS happy about something?